src/mongocxx/test/gridfs/bucket.cpp

```diff
@@ -1059,4 +1056,57 @@ TEST_CASE("gridfs download large file", "[gridfs::bucket]") {
 
     REQUIRE(total_bytes_read == length);
 }
+
+TEST_CASE("gridfs automatic index creation", "[gridfs::uploader] [gridfs::downloader]") {
+    using bsoncxx::builder::basic::document;
+    using bsoncxx::builder::basic::sub_array;
+
+    instance::current();
+    client client{uri{}};
+    database db = client["gridfs_index_creation"];
+    auto bucket = db.gridfs_bucket();
+
+    const size_t file_size = 100;
+    std::array<std::uint8_t, file_size> to_write, to_read;
+    std::iota(begin(to_write), end(to_write), 0);
+
+    auto uploader = bucket.open_upload_stream("test_file");
+
+    uploader.write(to_write.data(), file_size);
+    auto result = uploader.close();
+
+    auto downloader = bucket.open_download_stream(result.id());
+    auto bytes_read = downloader.read(to_read.data(), file_size);
+    REQUIRE(bytes_read == file_size);
+
+    SECTION("default indexes created") {
+        auto bucket_collection_name = GENERATE(values({"fs.chunks", "fs.files"}));
+        auto indexes = db[bucket_collection_name].list_indexes();
+        REQUIRE(std::distance(indexes.begin(), indexes.end()) == 2);
+    }
```

Comment:
So after uploading arbitrary data the default indexes are 
```
  { "v" : 2, "key" : { "_id" : 1 }, "name" : "_id_" }
  { "v" : 2, "unique" : true, "key" : { "files_id" : 1, "n" : 1 }, "name" : "files_id_1_n_1" }
```
This looks ok to me but weirdly, on line 1092, changing `_id` to `files_id` fails since the DB command doesn't match that index (`files_id`) with the one created here (`_id`) and doesn't throw an error. However, in C ([here](https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/tests/test-mongoc-gridfs-bucket.c#L183)) we test this using `files_id`. I just wanted to point this out because I'm not sure why this works in C but not here.